:tada: Improvements for wazuh importer

quirinziessler opened the pull request:
**Description**

This PR is a follow up for the closed PR #8746 re-written for file import. It includes all the features and adds a improvement on the pre-processing script which I published on a seperate repo for easier integration.

manuel-sommer commented on docs/content/en/integrations/parsers/file/wazuh.md:
```diff
@@ -2,4 +2,4 @@
 toc_hide: true
 ---
-Import JSON report.
+Import findings from Wazuh. The export from wazuh should be done via the script [available here](https://github.com/quirinziessler/wazuh-findings-exporter). The script fetches the findings by Wazuh client groups and saves them as json, ready for upload. Have in mind to adjust the max file size via "DD_SCAN_FILE_MAX_SIZE" if the file is larger than the default value of 100MB.
```
You introduce a new JSON format her which is generated as an additional component between the scanner and DefectDojo. This brings new complexity and the default scanner output can't be processed any more. 

I would suggest that you either include this code into this DefectDojo parser to be not dependant on this additonal script or advance this parser in a way to have two options to upload the output: 

1. Being able to parse the default JSON output directly from Wazuh
2. Using your exporter to generate the new JSON structure and parsing the JSON. 

Furthermore, please extend your testcoverage.

Maffooch replied:
@manuel-sommer excellent feedback! 

@quirinziessler The use of this extra script needs to be optional such that the original parser works with our without the preprocessing

quirinziessler commented:
Hi @Maffooch & @manuel-sommer - I updated the Parser so it can consume both types of files and also updated the unittests as you both recommended. Please review again.

manuel-sommer commented on docs/content/en/integrations/parsers/file/wazuh.md:
```diff
@@ -2,4 +2,10 @@
+- export the Wazuh findings from API and upload them to DefectDojo as json file. This method may be the easiest one but does export all known vulnerabilities at once. It is not possible to sort them after clients or any other categories. You will receive all vulnerabilities in one engagement.
+- export the findings via the script [available here](https://github.com/quirinziessler/wazuh-findings-exporter). The script fetches the findings by Wazuh client groups and saves them as json, ready for upload. You will receive one file per group allowing you to separate the clients via engagements in Wazuh. It also exports the endpoints hostname and displays them in DefectDojo UI.
+
+Independent of your above choice: Have in mind to adjust the max file size via "DD_SCAN_FILE_MAX_SIZE" if you see files larger than the default value of 100MB. Depending on the amount and category of integrated devices, the file size jumps rapidly.
```
This PR was merged in 2.30.2:https://github.com/DefectDojo/django-DefectDojo/pull/9271
Here you can see the relevant page:
https://documentation.defectdojo.com/contributing/how-to-write-a-parser/#update-the-import-page-documentation

Could you please extend your .md file in a way to e.g. ### Sample Scan Data and Link to the scanner.

Example see here: https://github.com/DefectDojo/django-DefectDojo/pull/9347

quirinziessler replied:
Sorry didn't have this in mind. Updated it according to the Guidelines.

Maffooch reviewed (approved):
@quirinziessler I did not consider a separate pypi package when thinking about this PR previously, but I think it's a great idea! Plus it gets your name out there :)

mtesauro reviewed (approved):
Approved

quirinziessler commented:
@grendel513 @cneill @devGregA @madchap do you maybe have some time to also approve this PR to be part of the next release? Many thanks!

falkowich commented:
Hi, 

Is this backward compatible with the "old" Wazuh parser?
I got some strange results when testing out dev and doing a "json"  import for each Wazuh agent as before.

![image](https://github.com/DefectDojo/django-DefectDojo/assets/1552705/b879d333-aa6d-4acc-a8d1-f533db72aea9)


I'm quite new to defectDojo, so it's a big possibility that this is just a skill issue :)

--
Kind Regards Falk

quirinziessler commented:
Hi @falkowich 
yeah it's backward compatible. Otherwise unittests for the original json output would been failing.
How does your json look like and how do you import it to DD? If you simply dump your api call to a .json file and import it to DD it should run fine.

falkowich commented:
Then I'll pledge "skill issues" for me :)

I'll export from wazuh and import in one script.
So perhaps I'll scramble the jsons somewhat before importing it.

I was testing dev out and I just saw that my tests got borked, thats why I asked.

If I get some more time with dev I can test some more, but until then just ignore this :)

--
Kind Regards Falk

quirinziessler commented:
If you look for a script for an export by agent_group - I wrote something for my purpose. Maybe it suits your needs as well.
https://github.com/quirinziessler/wazuh_findings_exporter
